src/applicationManager.ts

```diff
@@ -528,3 +546,239 @@ async function getGraphAPIRoles(token: string, graphApiPermissions: string[]) {
+  );
+  return assignmentExists;
+}
+
+/**
+ * Assigns groups to App Roles for a given application
+ * @param groupId Azure entra group ID to assign an app role to
+ * @param appRoleId ID of the app role to assign the group to
+ * @param applicationId ID of the application to assign the group to app roles for
+ */
+async function updateApplicationGroupAssignments({
+  token,
+  groupId,
+  appRoleId,
+  applicationId,
+}: {
+  token: string;
+  groupId: string;
+  appRoleId: string;
+  applicationId: string;
+}): Promise<void> {
+  const existingAssignment = await checkIfGroupAppRoleAssignmentExists({
+    token,
+    groupId,
+    appRoleId,
+    applicationId,
+  });
+
+  if (!existingAssignment) {
+    let body = {
+      principalId: groupId,
+      resourceId: applicationId,
+      appRoleId: appRoleId,
+    };
+    const result = await fetch(
+      `https://graph.microsoft.com/v1.0/groups/${groupId}/appRoleAssignments`,
+      {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(body),
+      },
+    );
+    console.log("Added group role assignment");
+    await errorHandler("updating application config", result);
+  } else {
+    console.log("Group role assignment already exists, skipping");
+  }
+}
+
+/**
+ * Updates an Azure Entra Application group assignments, specifically for app roles
+ * @param applicationId The object ID of enterprise application
+ * @param appRoles A collection (array) of AppRole custom objects to update the application with
+ */
+export async function addAppRoleGroupAssignmentsToApp({
+  token,
+  applicationId,
+  appRoles,
+}: {
+  token: string;
+  applicationId: string;
+  appRoles: AppRoles;
+}) {
+  for (const role of appRoles) {
+    for (const group of role.groups) {
+      let groupId = await getEntraGroupId(group, token);
+      await updateApplicationGroupAssignments({
+        token: token,
+        groupId: groupId,
+        appRoleId: role.id,
+        applicationId: applicationId,
+      });
+      console.log(
+        "Updated group role assignments for:",
+        role.displayName,
+        "App Role, processing group:",
+        group,
+      );
+    }
+  }
+}
+
+/**
+ * Updates an Azure Entra Application with a collection of App Roles
+ * @param applicationId The object ID of application registration
+ * @param appRoles A collection (array) of AppRole custom objects to update the application with
+ */
+export async function addAppRoles({
+  token,
+  applicationId,
+  appRoles,
+}: {
+  token: string;
+  applicationId: string;
+  appRoles: AppRoles;
+}) {
+  let appRolesCollection: Array<AppRole> = [];
+
+  // To keep existing functionality working of assigning AD groups general access
+  appRolesCollection.push({
+    allowedMemberTypes: ["User"],
+    description: "User",
+    displayName: "User",
+    id: "18d14569-c3bd-439b-9a66-3a2aee01d14f",
```

Comment:
can you comment on what this `id` is?

Reply:
Some GUID i took from the existing User role that seems consistent across other apps